DXRando/DeusEx/Classes/DXRBase.uc

```diff
@@ -174,25 +176,53 @@ simulated function bool RandoLevelValues(Actor a, float min, float max, out stri
     if( aug != None ) len = ArrayCount(aug.LevelValues);
     else if( sk != None ) len = ArrayCount(sk.LevelValues);
 
+    // figure out the range we will use
+    if( aug != None ) {
+        d_min = aug.Default.LevelValues[0];
+        d_max = aug.Default.LevelValues[len-1];
+    }
+    else if( sk != None ) {
+        d_min = sk.Default.LevelValues[0];
+        d_max = sk.Default.LevelValues[len-1];
+    }
+
+    // expand the range for more variety
+    avg_diff = (d_max - d_min) / float(len);
+    d_min -= avg_diff*min;
+    d_max += avg_diff*max;
+
+    // choose random points within the 0-1 range
     for(i=0; i < len; i++) {
-        if( aug != None ) d = aug.Default.LevelValues[i];
-        else if( sk != None ) d = sk.Default.LevelValues[i];
+        points[i] = rngf();
+        // some slight weighting
+        if( i == 0 ) points[i] *= 0.75;
+        if( i == len-1 ) points[i] = points[i]*0.75+0.25;
```

Comment:
instead of this weighting, maybe I could do like
`points[i] = ( points[i] + float( i / (len-1) ) ) / 2;`